tests/utils/streams/test_dummy.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+from snuba.utils.streams.dummy import DummyConsumer
+from snuba.utils.streams.types import ConsumerError, Message, Partition, Topic
+
+
+def test_working_offsets() -> None:
```

Comment:
This test only applies to the dummy consumer right now, but it'd be nice if this was parameterized to run against the Kafka consumer at some point (see comment above) to ensure all consumers maintain the same behavior when dealing with some of these edge cases.